pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -101,17 +103,50 @@ func (gce *GCECloud) AttachDisk(diskName string, nodeName types.NodeName, readOn
 	if readOnly {
 		readWrite = "READ_ONLY"
 	}
-	attachedDisk := gce.convertDiskToAttachedDisk(disk, readWrite)
 
 	mc := newDiskMetricContext("attach", instance.Zone)
-	attachOp, err := gce.service.Instances.AttachDisk(
-		gce.projectID, disk.Zone, instance.Name, attachedDisk).Do()
+	source := gce.service.BasePath + strings.Join([]string{
+		gce.projectID, "zones", disk.Zone, "disks", disk.Name}, "/")
+	attachOp, err := gce.getAttachDiskOp(
+		disk.Name, disk.Kind, disk.Zone, readWrite, source, diskTypePersistent, instance.Name)
 
 	if err != nil {
 		return mc.Observe(err)
 	}
 
-	return gce.waitForZoneOp(attachOp, disk.Zone, mc)
+	return gce.manager.WaitForZoneOp(attachOp, disk.Zone, mc)
+}
+
+// Returns a version agnostic AttachDisk operation.
+func (gce *GCECloud) getAttachDiskOp(
+	diskName string,
+	diskKind string,
+	diskZone string,
+	readWrite string,
+	source string,
+	diskType string,
+	instanceName string) (gceObject, error) {
+	if gce.AlphaFeatureGate.Enabled(GCEDiskAlphaFeatureGate) {
+		attachedDiskAlpha := &computealpha.AttachedDisk{
+			DeviceName: diskName,
+			Kind:       diskKind,
+			Mode:       readWrite,
+			Source:     source,
+			Type:       diskType,
+		}
+		return gce.manager.AttachDisk(
```

Comment:
Can AttachDisk() take in library-agnostic arguments, so that you can construct the library-specific types in AttachDisk() instead of here?